drivers/serial/uart_nrfx_uarte.c

```diff
@@ -0,0 +1,630 @@
+static int uarte_nrfx_fifo_read(struct device *dev,
+				u8_t *rx_data,
+				const int size)
+{
+	int num_rx = 0;
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+	const struct uarte_nrfx_data *data = get_dev_data(dev);
+
+	if (nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_ENDRX)) {
+		/* Clear the interrupt */
+		nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_ENDRX);
+
+		/* Receive a character */
+		rx_data[num_rx++] = (u8_t)data->rx_data;
+
+		nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STARTRX);
+	}
+
+	return num_rx;
+}
+
+/** Interrupt driven transfer enabling function */
+static void uarte_nrfx_irq_tx_enable(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	nrf_uarte_int_enable(uarte, NRF_UARTE_INT_ENDTX_MASK);
+}
+
+/** Interrupt driven transfer disabling function */
+static void uarte_nrfx_irq_tx_disable(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	nrf_uarte_int_disable(uarte, NRF_UARTE_INT_ENDTX_MASK);
+
+	/* If there is nothing to send, driver will save an energy
+	 * when TX is stopped.
+	 */
+	nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STOPTX);
+}
+
+/** Interrupt driven transfer ready function */
+static int uarte_nrfx_irq_tx_ready_complete(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	return nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_ENDTX);
+}
+
+static int uarte_nrfx_irq_rx_ready(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	return nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_ENDRX);
+}
+
+/** Interrupt driven receiver enabling function */
+static void uarte_nrfx_irq_rx_enable(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	nrf_uarte_int_enable(uarte, NRF_UARTE_INT_ENDRX_MASK);
+}
+
+/** Interrupt driven receiver disabling function */
+static void uarte_nrfx_irq_rx_disable(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	nrf_uarte_int_disable(uarte, NRF_UARTE_INT_ENDRX_MASK);
+}
+
+/** Interrupt driven error enabling function */
+static void uarte_nrfx_irq_err_enable(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	nrf_uarte_int_enable(uarte, NRF_UARTE_INT_ERROR_MASK);
+}
+
+/** Interrupt driven error disabling function */
+static void uarte_nrfx_irq_err_disable(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	nrf_uarte_int_disable(uarte, NRF_UARTE_INT_ERROR_MASK);
+}
+
+/** Interrupt driven pending status function */
+static int uarte_nrfx_irq_is_pending(struct device *dev)
+{
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	return ((nrf_uarte_int_enable_check(uarte,
+					    NRF_UARTE_INT_ENDTX_MASK) &&
+		 uarte_nrfx_irq_tx_ready_complete(dev))
+		||
+		(nrf_uarte_int_enable_check(uarte,
+					    NRF_UARTE_INT_ENDRX_MASK) &&
+		 uarte_nrfx_irq_rx_ready(dev)));
+}
+
+/** Interrupt driven interrupt update function */
+static int uarte_nrx_irq_update(struct device *dev)
```

Comment:
I think it should be nrfx instead of nrx

Reply:
fixed